src/app/modules/user-details/containers/user-profile/user-profile.component.ts

```diff
@@ -1,10 +1,229 @@
+            linkedIn: [this.userDetails.linkedIn, []],
+            country: [this.userDetails.countryId, [Validators.required]],
+            city: [this.userDetails.cityId, [Validators.required]],
+            availability: [this.userDetails.availability, []],
+            skills: ['', []],
+        });
+    }
+
+    openSkillDialog() {
+        let dialogRef = this.dialog.open(SkillsDialogComponent, { data: { skills: this.remainingSkills, addedSkills: this.userSkills } });
+        dialogRef.afterClosed().subscribe(result => {
+            if (result.data.save) {
+                this.remainingSkills = result.data.skills;
+                this.userSkills = result.data.addedSkills;
+                this.addedSkillIds = this.userSkills.map(skill => { return skill['skillId'] });
+                this.userForm.get("skills")?.setValue(this.addedSkillIds.toString());
+            }
+        });
+    }
+
+    countryChanged(event: any) {
+        this.userDetailsService.getCityList(Number(event)).subscribe({
+            next: (res) => {
+                this.cities = res.data;
+            }
+        });
+        this.userForm.get("city")?.setValue('');
+    }
+
+    loadCountry() {
+        this.userDetailsService.getCountryList().subscribe({
+            next: (res) => {
+                this.countries = res.data;
+            },
+            error: (err) => {
+            }
+        });
+    }
+
+    loadAvailability() {
+        this.userDetailsService.getAvailability().subscribe({
+            next: (res) => {
+                this.availabilities = res.data;
+            }
+        });
+    }
+
+    loadSkills() {
+        this.userDetailsService.getAllSkills().subscribe({
+            next: (res) => {
+                this.allSkills = res.data.map((skill: any) => { return <Skills>{ skillId: skill.skillId, skillName: skill.skillName } });
+            }
+        })
+    }
+
+    getUserDetails(userId: number) {
+        this.userDetailsService.getUserDetails(userId).subscribe({
+            next: (res) => {
+                this.userDetails = {
+                    userId: res.data.userId,
+                    name: res.data.firstName,
+                    surname: res.data.lastName,
+                    employeeId: res.data.employeeId,
+                    manager: res.data.manager,
+                    title: res.data.title,
+                    department: res.data.department,
+                    profileText: res.data.profileText,
+                    whyVolunteer: res.data.whyIVolunteer,
+                    countryId: res.data.countryId,
+                    cityId: res.data.cityId,
+                    availability: res.data.availability,
+                    linkedIn: res.data.linkedInUrl,
+                    avatar: res.data.avatar
+                }
+                if (res.data.avatar === null) {
+                    this.imageSrc = "../../../../../assets/Images/user-img-large.png"
+                }
```

Comment:
../../../../.. remove this it will work /assets/Images/user-img-large.png please check

Reply:
Changes Done